### PR TITLE
Comment input

### PR DESCRIPTION
**Pull request by eldiegod**
https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2411%3A217347

fixes: #2450

### CODE REVIEW

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx
Comment (rafalpawlow) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx]: Why do we need `any` here?

Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx]: because I'm using custom props in the story that are not part of `CommentInputProps`

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: You missed curly braces here, besides that, why do we need a delay here?

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: Hmm, I'm still unsure how I should treat this `data-` pattern. I'll just leave a comment here for now.

File: packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx]: Why did you remove these stories?

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx]: Correct me if I'm wrong but I'm pretty sure that you don't need this. If Container had been a label, the text area would just focus automatically.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:27]: If I remember correctly how it works, you can pass ref to the `useResizesObserver` options. You don't need to create another `measureRef`.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:70]: This would be needed if my comment above is true.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx]: again - we could remove this if the container had been the label.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:46]: onBlur on TextArea isn't enough?

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: I would avoid using `!important` as much as I can. I think we could avoid this by passing the prop to StyledCommentRow?

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: Same as above

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx]: Let's disable `onComment`, `onCancel` and `className` in controls

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx:17]: I would add an image URL to `memberAvatarUrl` as default, just to make this story a little prettier and make sure that this prop works.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:18]: Do we need all the CommentRowProps? I think the highlight is not needed

**Review by drillprop — CHANGES_REQUESTED** [review]
Looks really pretty! 
However, I found a couple of things that we could improve. 

In terms of functionality: 
 - You still can write inside the text area when it's processing. 
 - I don't think keyboard navigation is handled. https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2431%3A215801  We could omit the "confirm" scenario, we could do this during the integration PR, but I think "dismiss" scenario we should include in this PR.

**Comment by drillprop** [comment]
Just last thing - I found that when I tap on the information button on my smartphone I don't see any tooltip.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx
Comment (drillprop) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx]: So let's just add `CommentInputProps & { cancelButton: boolean }` it would be just easier to spot which prop isn't part of the `CommentInputProps`

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: > I don't think keyboard navigation is handled. https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2431%3A215801 We could omit the "confirm" scenario, we could do this during the integration PR, but I think "dismiss" scenario we should include in this PR.

actually "confirm" is already there on the `onKeyDown` just need integrations.

For the dismiss I'm not too sure what should happen, I press scape and the whole comment I'm writing gets deleted? can you clarify @toiletgranny

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx
Comment (drillprop) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.stories.tsx]: Maybe we could also add `description` to this particular control? I don't know how it works in a storybook - this is not super important,  just a suggestion.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (drillprop) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: I think you replied to the wrong comment 😄   Just saying - I hope you won't overlook this 👇🏻 
> You missed curly braces here, besides that, why do we need a delay here?

Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: oops, yeah meant to add a general comment; so I pretty much grabbed the anim from your like dislike component and the delay was there but in this case, don't think is needed

File: packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx]: thought it became redundant since `Comment` and `CommentInput` showcased commentRow in a better scenario

Comment (drillprop) [reply packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx]: Hmm I see, I would opt for not removing it although. I think it's good to know that we have a separate wrapper that can be used in two different contexts.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx]: yeah, we don't need focus here but we do need to setActive sadly

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:27]: I tried that at first but it gave be buggy results where it would stop measuring this way is how I got it to work

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx]: removed

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:46]: don't think so, clicking on the container even if its a label will cause it to open and close quickly because it blurs and focuses again

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx:18]: not 100% sure but I don't think it will matter too much if not

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (kdembler) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: Let's accept it for now and we will discuss during next dev sync

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (toiletgranny) [reply packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: @eldiegod @drillprop Yeah, I can see how this part of the documentation was a bit unclear. And to answer your question, pressing escape should not erase your input from the "Comment input" component. It should feel as a regular _blur_ of an input, meaning the "Comment input" component changes its state back to Default.

**Comment by toiletgranny** [comment]
@eldiegod Since Bartek and Rafał already requested some changes, can I ask you to let me know once these are addressed so that I can also jump in and do the design review of the implementation? It would be great if we could always request a review from the designer behind a particular feature! ❤️

File: packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx]: but this is a CommentRow, not a Comment. We have stories for comment already. This is a different thing.

File: packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentRow/CommentRow.stories.tsx]: same as above

**Review by drillprop — CHANGES_REQUESTED** [review]

**Comment by eldiegod** [comment]
> @eldiegod Since Bartek and Rafał already requested some changes, can I ask you to let me know once these are addressed so that I can also jump in and do the design review of the implementation? It would be great if we could always request a review from the designer behind a particular feature! ❤️

@toiletgranny yep you're right, I think it should be ready for u know once you are back 🤓

**Review by drillprop — APPROVED** [review]
LGTM 💪🏻  but let's wait for the final approval from Adam

**Comment by drillprop** [comment]
Last thing:
I just noticed that you forgot to add the `index.ts` file in the `CommentInput` directory.

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts
Comment (drillprop) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.styles.ts]: I'm getting an error from emotion: 
`The pseudo class ":nth-last-child" is potentially unsafe when doing server-side rendering. Try changing it to ":nth-last-of-type".`

**Review by drillprop — CHANGES_REQUESTED** [review]
Found another thing 😅

**Review by drillprop — APPROVED** [review]

**Review by toiletgranny — CHANGES_REQUESTED** [review]
This looks stunning! 🔥  A few minor issues I found:

1. The bottom border color should be set to `var(--color-border-muted-alpha)`; currently, it's var`(--color-core-neutral-800-lighten)`.
2. On hover, the bottom border color should change to `var(--color-border-alpha)`; currently, it's not happening.
3. Ideally, the bottom border color transition is animated using the same timing and easing as everything else in the Comment input component, which would be `var(--animation-transition-medium)`; currently, it's `var(--animation-transition-fast)`.
4. Ideally, the tooltip component is positioned _above_ the Information component, and aligned to the left. Currently, it's placed below the Information component, and aligned to the right side.
5. Not sure if this issue belongs here or if it's just a misuse of the Tooltip component, but it seems like we're missing the max-width in the Tooltip component (the one about comments being stored on the blockchain). Notice how the body text is wrapped in the designs and how it's not the case in the implementation. CC @rafalpawlow.

**Comment by toiletgranny** [comment]
Much better! The bottom border is not quite there yet, however. Upon hovering over the Comment input, instead of changing the background-color property in the `.css-uja03m` class, you set a new background-color property on the `.css-6etn8w` class, which results in:
1. The lack of animation in the border color change,
2. Having a much brighter border color on hover, than anticipated (because the two transparent borders are now overlapping each other: the first one `var(--color-border-muted-alpha)` in the `.css-uja03m` class, and the second one `var(--color-border-alpha)`, in the `.css-6etn8w` class).

Once this minor issue is resolved, it's good to go from my perspective!

**Pull request merged**
